src/llm/http_llm_calculator.cc

```diff
@@ -46,29 +47,75 @@ class OpenAIChatCompletionsRequest {
     chat_t messages;
     bool stream{false};
     std::string model;
-    float frequencyPenalty{0.0f};
-    std::optional<int64_t> maxTokens{std::nullopt};
-    float presencePenalty{0.0f};
-    float temperature{1.0f};
-    float topP{1.0f};
+    std::optional<int> maxTokens{std::nullopt};
+    // float frequencyPenalty{0.0f};
+    // float presencePenalty{0.0f};
+    std::optional<float> diversityPenalty{std::nullopt};
+    std::optional<float> repetitionPenalty{std::nullopt};
+    std::optional<float> lengthPenalty{std::nullopt};
+    std::optional<int> numReturnSequences{std::nullopt};
+    std::optional<float> temperature{std::nullopt};
+    std::optional<float> topP{std::nullopt};
+    std::optional<int> topK{std::nullopt};
+    std::optional<int> bestOf{std::nullopt};
+    // std::optional<bool> useBeamSearch{std::nullopt};
+    std::optional<bool> ignoreEOS{std::nullopt};
 
 public:
     OpenAIChatCompletionsRequest(Document& doc) :
         doc(doc) {}
 
+    GenerationConfig createGenerationConfig() const {
+        GenerationConfig config;
+
+        // Generic
+        if (maxTokens.has_value())
+            config.max_new_tokens = maxTokens.value();
+        // TODO: max_length = ?
+        if (ignoreEOS.has_value())
+            config.ignore_eos = ignoreEOS.value();
+
+        // Beam search specific
+        config.num_groups = 1;  // OpenAI hardcoded
+        if (bestOf.has_value())
+            config.group_size = bestOf.value();
+        if (diversityPenalty.has_value())
+            config.diversity_penalty = diversityPenalty.value();  // TODO: Not available in OpenAI nor vLLM
+        // TODO: stop_criteria = ?
+        if (numReturnSequences.has_value())
+            config.num_return_sequences = numReturnSequences.value();
+        if (repetitionPenalty.has_value())
+            config.repetition_penalty = repetitionPenalty.value();
+        if (lengthPenalty.has_value())
+            config.length_penalty = lengthPenalty.value();
+        // TODO: no_repeat_ngram_size = ?
+        // TODO: early_finish = ?
+        // TODO use_beam_search is unused ?
+
+        // Multinomial specific
+        if (temperature.has_value())
+            config.temperature = temperature.value();
+        if (topK.has_value())
+            config.top_k = topK.value();
+        if (topP.has_value())
+            config.top_p = topP.value();
+        config.do_sample = config.temperature > 0.0f && config.group_size == 1;
+        // TODO: rng_seed = ?
+
+        // TODO: bos_token_id = ?
+        // TODO: pad_token_id = ?
+        // TODO: eos_token_id = ?
```

Comment:
This can we read from tokenizer config. Not a generation config parameter.

Reply:
removed